### PR TITLE
Devise auth

### PR DESCRIPTION
**Pull request by bthooper**
Devise added.

I am not sure this does all that we need at the moment, but the basics are there- a user model, registration and login. We need to connect the "forgot password" feature to an email service.

Also, we now need to connect users to food resources. I assume this is a users have_many food resources and a food resource belongs_to one user?  That will be a new branch, so I wanted to go ahead and submit this with the devise functionality.

### CODE REVIEW

**Comment by thomascchen** [comment]
Thanks @bthooper! This all sounds right. Will try to take a look at this PR asap

File: test/models/user_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (thomascchen) [inline test/models/user_test.rb:4]: We can leave this as is for now. I'd love to add [rspec](https://github.com/rspec/rspec-rails) sometime.

File: app/controllers/users/confirmations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::ConfirmationsController < Devise::ConfirmationsController
```
Comment (thomascchen) [inline app/controllers/users/confirmations_controller.rb]: I don't think we need the confirmable module. We can probably delete this file.

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
```
Comment (thomascchen) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Same here, I don't think we need this. Can delete.

File: app/controllers/users/unlocks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::UnlocksController < Devise::UnlocksController
```
Comment (thomascchen) [inline app/controllers/users/unlocks_controller.rb]: Not sure we really need the lockable module either. I'd remove this.

File: db/migrate/20201020183830_devise_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      t.datetime :remember_created_at
+
+      ## Trackable
+      # t.integer  :sign_in_count, default: 0, null: false
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.inet     :current_sign_in_ip
+      # t.inet     :last_sign_in_ip
+
+      ## Confirmable
```
Comment (thomascchen) [inline db/migrate/20201020183830_devise_create_users.rb:24]: Oh yea confirmable and lockable stuff is commented out here, so yea we don't need those associated controllers.

**Review by thomascchen — CHANGES_REQUESTED** [review]
This is great! I think there's a few files we can remove. We can figure out an email service later. I'll add an issue for it. I think we might have some AWS credits sponsored by Code for America that we could use maybe.

I agree the next would be to establish a user have_many food resources relationship. 👍

**Comment by bthooper** [comment]
So, i deleted the unneeded controllers and pushed my commit up to my fork....but being a git novice, how do i get my commit to show up in this pull request? My googling research suggested that all I need to do is commit my changes....Heck, it even says "add more commits by pushing to the devise_auth branch on bthooper/food_resources' right above..and i did...and it shows in my repo...but why not here? am I missing something???

**Review by thomascchen — APPROVED** [review]
awesome!

**Pull request merged**